### PR TITLE
fix: var scope calculation

### PR DESCRIPTION
**Pull request by zeniroy**
declaration with var keyword has function scope. additional process needed

### CODE REVIEW

**Comment by nchanged** [comment]
@zeniroy amazing job! But can you cover it with tests? It's hard to follow what's the issue, so the tests would actually help!

**Comment by nchanged** [comment]
@zeniroy `src/compiler/__tests__/scopes.test.ts`

**Comment by zeniroy** [comment]
ok i will add some test soon.

**Review by nchanged — APPROVED** [review]
👍
